components/new_post/post_component.tsx

```diff
@@ -0,0 +1,333 @@
+import InfoSmallIcon from 'components/widgets/icons/info_small_icon';
+import PostPreHeader from 'components/post_view/post_pre_header';
+import EditPost from 'components/edit_post';
+import AutoHeightSwitcher, {AutoHeightSlots} from 'components/common/auto_height_switcher';
+import {Props as TimestampProps} from 'components/timestamp/timestamp';
+
+import PostUserProfile from './user_profile';
+import PostOptions from './post_options';
+
+export type Props = {
+    post: Post;
+    teamId: string;
+    currentUserId: string;
+    compactDisplay?: boolean;
+    colorizeUsernames?: boolean;
+    isFlagged: boolean;
+    isBusy?: boolean;
+    previewCollapsed?: string;
+    previewEnabled?: boolean;
+    isEmbedVisible?: boolean;
+    enableEmojiPicker?: boolean;
+    enablePostUsernameOverride?: boolean;
+    isReadOnly?: boolean;
+    pluginPostTypes?: {[postType: string]: PostPluginComponent};
+    channelIsArchived?: boolean;
+    isConsecutivePost?: boolean;
+    isLastPost?: boolean;
+    recentEmojis: Emoji[];
+    handleCardClick?: (post: Post) => void;
+    togglePostMenu?: (opened: boolean) => void;
+    a11yIndex?: number;
+    isBot: boolean;
+    actions: {
+        markPostAsUnread: (post: Post, location: string) => void;
+        emitShortcutReactToLastPostFrom: (emittedFrom: 'CENTER' | 'RHS_ROOT' | 'NO_WHERE') => void;
+        setActionsMenuInitialisationState: (viewed: Record<string, boolean>) => void;
+        selectPost: (post: Post) => void;
+        removePost: (post: Post) => void;
+    };
+    timestampProps?: Partial<TimestampProps>;
+    shouldHighlight?: boolean;
+    isPostBeingEdited?: boolean;
+    isMobileView: boolean;
+};
+
+const PostComponent = (props: Props): JSX.Element => {
+    const postRef = useRef<HTMLDivElement>(null);
+    const postHeaderRef = useRef<HTMLDivElement>(null);
+    const [hover, setHover] = useState(false);
+    const [a11yActive, setA11y] = useState(false);
+    const [fileDropdownOpened, setFileDropdownOpened] = useState(false);
+    const [alt, setAlt] = useState(false);
+
+    useEffect(() => {
+        if (postRef.current) {
+            postRef.current.addEventListener(A11yCustomEventTypes.ACTIVATE, handleA11yActivateEvent);
+            postRef.current.addEventListener(A11yCustomEventTypes.DEACTIVATE, handleA11yDeactivateEvent);
+        }
+        if (a11yActive) {
+            postRef.current?.dispatchEvent(new Event(A11yCustomEventTypes.UPDATE));
+        }
+    }, []);
+
+    useEffect(() => {
+        if (hover) {
+            removeKeyboardListeners();
+        }
+
+        if (postRef.current) {
+            postRef.current.removeEventListener(A11yCustomEventTypes.ACTIVATE, handleA11yActivateEvent);
+            postRef.current.removeEventListener(A11yCustomEventTypes.DEACTIVATE, handleA11yDeactivateEvent);
+        }
+    }, [hover]);
+
+    const getClassName = (post: Post, isSystemMessage: boolean, isMeMessage: boolean) => {
```

Comment:
Minor thing, perhaps we could move the call to `isMeMessage` into here instead of calling it outside of this and passing it in? If we inline it in the call to `classNames`, we can get rid of the naming conflict as well